imports/plugins/core/accounts/client/components/addressBookGrid.js

```diff
@@ -0,0 +1,224 @@
+import React, { Component } from "react";
+import PropTypes from "prop-types";
+import { Components, registerComponent } from "@reactioncommerce/reaction-components";
+
+class AddressBookGrid extends Component {
+  static propTypes = {
+    /**
+     * array of address objects
+     */
+    addressBook: PropTypes.arrayOf(PropTypes.shape({
+      _id: PropTypes.String,
+      fullName: PropTypes.String,
+      address1: PropTypes.String,
+      addresss2: PropTypes.String,
+      postal: PropTypes.String,
+      city: PropTypes.String,
+      region: PropTypes.String,
+      country: PropTypes.String,
+      phone: PropTypes.String,
+      isBillingDefault: PropTypes.Bool,
+      isShippingDefault: PropTypes.Bool,
+      isCommercal: PropTypes.Bool
+    })),
+    /**
+     * selects an address to be edited and renders the AddressBookForm
+     */
+    edit: PropTypes.func,
+    /**
+     * removes the selected address
+     */
+    remove: PropTypes.func,
+    /**
+     * selects a default shipping or billing address
+     */
+    select: PropTypes.func
+  }
+
+  state = {
+    defaultShippingAddressId: this.defaultShippingAddressId,
+    defaultBillingAddressId: this.defaultBillingAddressId
+  }
+
+  componentWillReceiveProps(nextProps) {
+    let { addressBook } = nextProps;
+    if (!Array.isArray(addressBook)) addressBook = [];
+    // if a new address has been added to the addressBook
+    // check to see if the new address is now the default shipping or billing address
+    addressBook.forEach((addy) => {
+      if (addy.isShippingDefault && addy._id !== this.defaultShippingAddressId) {
+        this.setDefaultAddress(addy._id, "shipping");
+      }
+
+      if (addy.isBillingDefault && addy._id !== this.defaultBillingAddressId) {
+        this.setDefaultAddress(addy._id, "billing");
+      }
+    });
+  }
+
+  // Address Book Grid helpers
+
+  /**
+   * @method setDefaultaddress
+   * @summary updating the compoenet state with the new default shipping or billing address.
+   * @since 2.0.0
+   * @param {String} _id - address object _id.
+   * @param {String} usage - the address usage "shipping" or "billing".
+   */
+  setDefaultAddress(_id, usage) {
+    let { defaultBillingAddressId, defaultShippingAddressId } = this.state;
+    if (usage === "shipping") defaultShippingAddressId = _id;
+    if (usage === "billing") defaultBillingAddressId = _id;
+    this.setState({ defaultBillingAddressId, defaultShippingAddressId });
+  }
+
+  /**
+   * @method defaultShippingAddressId
+   * @summary getter that returns ether the default shipping address _id or and empty string.
+   * @since 2.0.0
+   * @return {String} - default shipping address _id or empty string.
+   */
+  get defaultShippingAddressId() {
+    const { addressBook } = this.props;
```

Comment:
Since you're calling `.find` blindly on this, I recommend the same `this.addressBook` getter pattern you did in the other component.